### ISSUE TITLE
pattern broken from 3.x to 4.x

### ISSUE THREAD

**Issue by fringd** [micromatch/micromatch#179]
_(Thanks for reporting an issue to micromatch! If you haven't already read the [contributor guidelines](contributing.md), Please do that now, then proceed to fill out the details below.)_

## Please describe the **minimum necessary steps** to reproduce this issue:
```js
const mm = require('micromatch');
const files = ['yoooo.js','foo/baz/something.js', 'bar/baz/something.js', 'anything-else/foo.js', 'anything/__tests__/else.js'];
mm(files, '!(bar/baz|foo|**/__tests_)/**/*.js')
```

## What is happening that shouldn't be?
```js
[ 'foo/baz/something.js',
  'bar/baz/something.js',
  'anything-else/foo.js',
  'anything/__tests__/else.js' ]
```

all the files match. this is with 4.0.2

## What should be happening instead?
```
[ 'anything-else/foo.js' ]
```
this worked correctly in 3.x

**Comment by Ignigena** [comment]
Seeing similar things here with a negative matching glob and was able to reproduce on RunKit https://npm.runkit.com/micromatch

```
const micromatch = require("micromatch@3.1.10")
micromatch([
  'tests/fixtures/sns/notification/delete.js'
], [
  '**/*.js',
  '!(coverage|config|scripts|tests)/**'
])
```

Returns `[]` as expected since the file path is excluded by the second glob pattern. However, in version 4, this exact same code is broken and returns the file path as a match:

```
const micromatch = require("micromatch@4.0.2")
micromatch([
  'tests/fixtures/sns/notification/delete.js'
], [
  '**/*.js',
  '!(coverage|config|scripts|tests)/**'
])
```

If I update the second glob pattern to `!**/(coverage|config|scripts|tests)/**` the path is excluded in version 4 but it feels like a bit of a workaround since the pattern was written that way to only exclude when the root directory matches.

**Comment by jonschlinkert** [comment]
@Ignigena thanks for confirming. I'll look into this.

**Comment by SimenB** [comment]
@jonschlinkert were you able the look into this - it seemingly is a cause of a horrible performance regression in jest (see https://github.com/facebook/jest/issues/9457)

**Comment by csvan** [comment]
Any updates on this? Can I help?

**Comment by jonschlinkert** [comment]
@csvan are you experiencing this issue as well? If so, any additional information would be useful.

I'm working on this. It's at the top of my list.

**Comment by csvan** [comment]
@jonschlinkert sorry for missing to reply on this. I'm part of the Jest crowd experiencing a severe performance regression over this (see https://github.com/facebook/jest/issues/9457)

**Comment by stramel** [comment]
@jonschlinkert I can confirm this as well. Prettier has an extensive test suite that I ran it against. https://github.com/prettier/prettier/pull/7726

The case was:

Pattern: `**/__best-tests__/*.js`
File Path: `__best-tests__/file.js`
Marked as not a match 

https://github.com/prettier/prettier/blob/61d3936b6e13238e0eb16fab3ca53d75fc942030/tests_integration/__tests__/config-resolution.js#L70

All tests pass with v3.1.10
4 tests fail with v4.0.2

**Comment by csvan** [comment]
Hi @jonschlinkert, are there any news on this issue?

**Comment by tylersmalley** [comment]
This is also a blocker for us, so stopping by to show my support ❤️

**Comment by kibertoad** [comment]
How does fixing this in picomatch address the problem in micromatch?..

**Comment by jonschlinkert** [comment]
Hi @kibertoad, micromatch depends on [picomatch](https://github.com/micromatch/micromatch/blob/master/package.json#L41).

**Comment by danez** [comment]
This will be fixed once picomatch gets updated here. Leaving open for now.

**Comment by kibertoad** [comment]
@jonschlinkert Thank you for clarificaiton! Wouldn't it make sense to bump dependency version on picomatch here then?

**Comment by danez** [comment]
The regression is now fixed in 4.0.4, but I'm not certain it fixes the performance issue mentioned in jest. If the performance issue still persists please open a separate issue with a repro case.
